### PR TITLE
feat: create InputDatePicker component

### PR DESCRIPTION
**Pull request by kimak**
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

### CODE REVIEW

File: packages/streamline/src/components/input-date-picker/input-date-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    InputDatePickerProps,
+    'maximumDate' | 'minimumDate' | 'okLabelIOS'
+  > {
+  focused: boolean;
+  setFocused: (focused: boolean) => void;
+  innerOnChange?: () => void;
+  innerDate?: Date;
+}
+
+const ModalPickerIOS = ({
```
Comment (hcourthias) [inline packages/streamline/src/components/input-date-picker/input-date-picker.tsx]: Could you extract this component to another file ?

File: packages/streamline/src/components/input-date-picker/input-date-picker.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      setFocused(false);
+    }}
+  >
+    <Pressable
+      accessibilityRole="button"
+      accessibilityLabel={okLabelIOS.toLocaleLowerCase()}
+      onPress={() => {
+        setFocused(false);
+      }}
+      style={{ flex: 1 }}
```
Comment (hcourthias) [inline packages/streamline/src/components/input-date-picker/input-date-picker.tsx]: Avoid inline style

File: packages/streamline/src/components/input-date-picker/input-date-picker.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      </Box>
+      <DateTimePicker
+        testID="date-picker-input"
+        value={innerDate || new Date()}
+        display="inline"
+        onChange={innerOnChange}
+        minimumDate={minimumDate}
+        maximumDate={maximumDate}
+        themeVariant="light"
+        style={{ height: 400 }}
```
Comment (hcourthias) [inline packages/streamline/src/components/input-date-picker/input-date-picker.tsx]: Same here, Avoid inline style

**Review by hcourthias — CHANGES_REQUESTED** [review]

File: packages/sandbox/e2e/ios/screenshots/InputDatePicker.png
Comment (kimak) [inline packages/sandbox/e2e/ios/screenshots/InputDatePicker.png:1]: @ianrobertlambert to review on iOS

File: packages/sandbox/e2e/android/screenshots/InputDatePicker.png
Comment (kimak) [inline packages/sandbox/e2e/android/screenshots/InputDatePicker.png:1]: @ianrobertlambert to review on Android

**Review by judastheo — APPROVED** [review]

**Review by hcourthias — APPROVED** [review]

**Comment by ianrobertlambert** [comment]
Thanks @kimak ! Sorry for my late reply on this.
Reviewed! Both iOS and Android are working well. 😊✨

- Can we please update the blue colors on both iOS and Android to match our Bluko brand color?
Bluko-500: `#2D50E6`

- Just so I better understand... Are we using a "library" for this? Will this mean we can stay aligned with the native behavior if the native picker on iOS/Android changes?

**Comment by kimak** [comment]
> Thanks @kimak ! Sorry for my late reply on this. Reviewed! Both iOS and Android are working well. 😊✨
> 
> * Can we please update the blue colors on both iOS and Android to match our Bluko brand color?
>   Bluko-500: `#2D50E6`

I updated it for iOS here: https://github.com/GetLuko/streamline/pull/85#discussion_r1228115074

On Android, I'm limited with expo the solution we use here for the sandbox
https://expo.canny.io/feature-requests/p/customize-android-apptheme

But it will be good once integrated with the final app:

Screenshot_20230613-145523_Staging


> * Just so I better understand... Are we using a "library" for this? Will this mean we can stay aligned with the native behavior if the native picker on iOS/Android changes?

Yes, it's a react-native library https://github.com/react-native-datetimepicker/datetimepicker using Native implementation.
It will be retro compatible with new OS versions. With that being said, sometime we still need update the lib to be up to date with the latest OS (last time was iOS 14 for example).

**Comment by ianrobertlambert** [comment]
Thanks for the clarity! Sounds good. 👍

**Pull request merged**